src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -33,9 +33,32 @@ public List<BookDto> findAll() {
 
     @Override
     public BookDto findById(long id) {
-        Book book = bookRepository.findById(id).orElseThrow(
+        Book book = findBookByIdOrThrow(id);
+        return bookMapper.toDto(book);
+    }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto createBookRequestDto) {
+        if (!bookRepository.existsById(id)) {
+            throw new EntityNotFoundException("Cant update book by id: " + id);
+        }
```

Comment:
could still be a private method like `validateExistanse(id)` to not duplicate the code